packages/gatsby-plugin-page-creator/src/collection-extract-query-string.ts

```diff
@@ -37,9 +38,16 @@ export function collectionExtractQueryString(
             if (!text) return
 
             if (text.includes(`...CollectionPagesQueryFragment`) === false) {
-              throw new Error(
-                `Your collection graphql query is incorrect. You must use the fragment "...CollectionPagesQueryFragment" to pull data nodes`
+              reporter.error(
+                `PageCreator: Your collection graphql query is incorrect. You must use the fragment "...CollectionPagesQueryFragment" to pull data nodes
+
+Please fix the query from ${absolutePath.split(`src/pages`)[1]}.
+
+Offending query: ${text}`
               )
+              queryString = ``
+              modelType = ``
```

Comment:
Isn't this unnecessary?

Reply:
Nope, it's so that the below query creation fails and returns an empty string, which is then handled by the parent function to stop execution. feels a bit weird but it's just the method for killing execution and allowing the user to fix their issues